### PR TITLE
SQLx refactor core changes

### PR DESCRIPTION
**Pull request by okrplay**
Not all routes are refactored yet, but I made multiple changes to core files like `main.rs` and want to get those onto the dev branch to prevent merge conflicts later on in case anyone works directly on the dev branch till then. I will merge the remaining changes in a later pull request.

### CODE REVIEW

File: sqlx-data.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          "ordinal": 2,
+          "name": "password",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 3,
+          "name": "created",
+          "type_info": "Timestamptz"
+        }
+      ],
```
Comment (cloudybyte) [inline sqlx-data.json:27]: Please add a column for the avatar url

File: migrations/1_create_users.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE users (
+    id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+    name text NOT NULL UNIQUE,
+    password text NOT NULL,
+    created timestamp with time zone NOT NULL DEFAULT current_timestamp
+)
```
Comment (oleggtro) [inline migrations/1_create_users.sql:6]: Please add a column for the avatar url

**Review by oleggtro — CHANGES_REQUESTED** [review]
Please add the avatar url column and this thing is good to go

**Comment by okrplay** [comment]
I'll add that and the other missing things in the next and final PR from `sqlx-refactor` to `dev`.

File: migrations/1_create_users.sql (hunk 2 above)
Comment (okrplay) [reply migrations/1_create_users.sql:6]: I'll add that in the next and final PR from `sqlx-refactor` to `dev`.

**Review by oleggtro — APPROVED** [review]
approved

**Comment by okrplay** [comment]
The remaining user functions have been implemented in de68b1b092c81409a257a5801643350d8cefaf06 and the issue #8 has been created to keep track of the implementation of user icon and user status.

**Pull request merged**